### PR TITLE
172 Add more data to the PDF

### LINKED ISSUE #172: Prepare PDF with sample data
As a user I want to see a pdf with data about registration such that I know how to add the card in the app.

Acceptance Criteria:
  - [ ] Include a how to 
  - [ ] Include QR code

### PR DESCRIPTION
**Pull request by maxammann**
fixes #172

### CODE REVIEW

File: administration/src/util/qrcode.ts [hunk 1]
```diff
@@ -64,22 +69,33 @@ const createQRCode = (text: string, renderRect: (x: number, y: number, size: num
...
     const hints = new Map()
     hints.set(EncodeHintType.MARGIN, 0)
     hints.set(EncodeHintType.QR_VERSION, 7) // 224 characters
 
     let pathData = ""
-    createQRCode(text, (x: number, y: number, size: number) => pathData += 'M' + x + ',' + y + ' V' + size + ' H' + size + ' V' + y + ' H' + x + ' Z '
-        , size,
-        hints)
+    createQRCode(text, (x: number, y: number, size: number) => {
+            return pathData += 'M' + x + ',' + y + ' V' + size + ' H' + size + ' V' + y + ' H' + x + ' Z ';
```
Comment (Schedulaar) [inline administration/src/util/qrcode.ts:95]: `Z` already means: Close the path, hence you can omit the last H:
Also you must use lower `h` / `v` to make it a relative horizontal/vertical line (move pen `size` units to the right vs. move pen to x-coordinate `size`) I did not test the following:
```suggestion
            return pathData += 'M' + x + ',' + y + ' v' + size + ' h' + size + ' v' + (-size) + ' Z ';
```

**Review by Schedulaar — APPROVED** [review]

File: administration/src/components/logo.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
...K5XtnVLpVAwAey9AHlmARWFnhf93H9QaI48Pwhhxwr++MkNrl67QyKeeJMEisKlC11486wdVDJpIvPDtlElJZG5Qc48+3zfjb5yoiBTXkqJpqnZUvA9RlQIlhaWuPbl18RiibJ19diIFdy4x/zCMv/w9z8hXBey29FbFq2tjbS1t9DbO4DYo9BTCkHTZA+KlUax0kTmBivevLnSKDgOoGsahmnsKX8rk+HmrW8ZG3NW9e8EIQRSSh49es783CK/+MVPOPnOURACw2XyTmcHfX2D+Q1IOpGN71Oxlu37icKXAF1D3ysNrAiedvdlVX8FW58LwdDwBP/66z9y9dodlpdWAEFTYx1qAdXLm1u5/rqFD0VoAE3T3t4TQBHMTM3ba3IiWfFzgRVFsLYW5eq1O3R393H69AkWFpaxijng+hCgQBsAVFXZvVOHECRiCb64epupqbmKdPPa+THs+05MzjA+Mc32Tmg1bKKoLMKuYyklX915yNPuvn0T/pvPWRP8XijMBhB2T6Ad+wIJwdPuXm7e+rYifYNqcAYFEUAAyWSKVGpb9klRGBwc49NPbxGLxWuz7gChQC9AEI3GWV2Nbq4DisLoyAR/+L9XmSupILOG/UBhGkBAPJ6gf2AEsANDvS/6+e3vPmNiorDcew3VgaIKQm5/9ZBYLMHa2jrdz16xvh6tCf+AomACCCFYWlrh2pdfY7eTr1naBxlFuoEiawLUBH/QsT+tMGuoGtQIcMhRI8AhR40Ahxw1Ahxy1AhwyFEjwCFHjQCHHDUCHHLUCHDIUSPAIUeNAIccNQIccvx/7JyF+GroeEoAAAA4ZVhJZk1NACoAAAAIAAGHaQAEAAAAAQAAABoAAAAAAAKgAgAEAAAAAQAABACgAwAEAAAAAQAABAAAAAAA8BXIJwAAACV0RVh0ZGF0ZTpjcmVhdGUAMjAyMS0wMS0wNlQxODoxNzoxOCswMDowMITcU9UAAAAldEVYdGRhdGU6bW9kaWZ5ADIwMjEtMDEtMDZUMTg6MTc6MTgrMDA6MDD1getpAAAAEnRFWHRleGlmOkV4aWZPZmZzZXQAMjZTG6JlAAAAGXRFWHRleGlmOlBpeGVsWERpbWVuc2lvbgAxMDI08sVWHwAAABl0RVh0ZXhpZjpQaXhlbFlEaW1lbnNpb24AMTAyNEs+jfcAAAAASUVORK5CYII='
```
[the remark this answers is not in the document]
Comment (maxammann) [reply administration/src/components/logo.ts:3]: That is a png :D just encoded as base64. There is no simple way to use webpack for this with a simple setup like ours.

File: administration/src/components/PDFView.tsx [hunk 3]
```diff
@@ -1,23 +1,59 @@
...
+
+        const qrCodeSize = 100;
+        const qrCodeY = pageBottom - qrCodeSize - 40;
+        const qrCodeMarginTop = 5
+        doc.setFontSize(16)
+        doc.text("Anmeldecode:", pageMargin, qrCodeY - qrCodeMarginTop);
+        drawjsPDF("A".repeat(223), pageMargin, qrCodeY, qrCodeSize, doc)
+
+        doc.setFontSize(12)
+        doc.textWithLink("https://ehrenamtskarte.app/download_app", width - pageMargin - 80, pageBottom, { url: "https://ehrenamtskarte.app/download_app" });
```
[the remark this answers is not in the document]
Comment (maxammann) [reply administration/src/components/PDFView.tsx:49]: The first option is simpler :grimacing:

File: administration/src/util/qrcode.ts (hunk 1 above)
Comment (maxammann) [reply administration/src/util/qrcode.ts:95]: I present the H.S.O.:

  Human SVG Optimizer

File: administration/src/components/logo.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Schedulaar) [reply administration/src/components/logo.ts:3]: Hmm... what happens if you have it as a normal png? I though webpack would do that itself :thinking: I mean you can simply do `` in react and I think usually this emits a base-64 source?
But yeah, doesn't really matter

[the remark this answers is not in the document]
Comment (maxammann) [reply administration/src/components/logo.ts:3]: It potentially could output a path to the PNG instead oft the PNG.

**Review by aiquita — APPROVED** [review]

**Pull request merged**